src/lib/paginator/paginator-intl.ts

```diff
@@ -38,19 +38,41 @@ export class MatPaginatorIntl {
   lastPageLabel: string = 'Last page';
 
   /** A label for the range of items within the current page and the length of the whole list. */
-  getRangeLabel = (page: number, pageSize: number, length: number) => {
-    if (length == 0 || pageSize == 0) { return `0 of ${length}`; }
+  getRangeLabel =
+      (page: number, pageSize: number, length: number) => {
+        if (length == 0 || pageSize == 0) {
+          return `0 of ${length}`;
+        }
+
+        length = Math.max(length, 0);
+
+        const startIndex = page * pageSize;
+
+        // If the start index exceeds the list length, do not try and fix the end index to the end.
+        const endIndex =
+            startIndex < length ? Math.min(startIndex + pageSize, length) : startIndex + pageSize;
+
+        return `${startIndex + 1} - ${endIndex} of ${length}`;
+      }
+
+  /**
+   * Screenreader friendly label for the range of items within the current page and length of the
+   *  whole list.
+   */
+  getReaderRangeLabel = (page: number, pageSize: number, length: number) => {
```

Comment:
This should be an optional function. If it doesn't exist, the `reader` element could just use the `getRangeLabel` function